scvid/data/dadc_dataset.py

```diff
@@ -171,6 +182,29 @@ def __iter__(self):
             # clear lru cache
             self.dadc.cache.clear()
 
+        # replicas
+        rank, num_replicas = get_rank_and_num_replicas()
```

Comment:
In my experience, code involving ceiling function, //, is a bug attractor :D I will be sure to look at your test cases more carefully than the actual code...